rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -49,6 +49,30 @@ void SequentialReader::reset()
   storage_.reset();
 }
 
+void SequentialReader::open_storage()
+{
+  storage_ = storage_factory_->open_read_only(
+    *current_file_iterator_, metadata_.storage_identifier);
+  if (!storage_) {
+    throw std::runtime_error("No storage could be initialized. Abort");
+  }
+}
+
+void SequentialReader::setup_compression()
```

Comment:
No unit test for this method?

Reply:
Since this is part of the `open()` method and we already check if the proper compression methods are called `decompress_file/message()` then I don't think this needs a unit test.

Reply:
Per hallway discussion, one thing we could do is pass bad compression options and make sure the `open()` method throws.